File: be_golang/models/transaction.go
package models

import "time"

type Transaction struct {
	IdTransaction int    `json:"id_transaction" gorm:"primaryKey;type:int;unique;not null"`
	AmountDiamond int    `json:"amount_diamond" gorm:"type: int"`
	TotalPrice    int    `json:"total_price" gorm:"type: int"`
	Status        string `json:"status" gorm:"type: varchar(255)"`
	UserId        int    `json:"user_id"`
	// UserResponse    User      `json:"user"`
	IdDiamond int `json:"id_diamond"`
	// DiamondResponse Diamond   `json:"diamonds"`
	Diamond   Diamond   `gorm:"foreignKey:IdDiamond"`
	User      User      `gorm:"foreignKey:UserId"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: be_laravel/app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Transaction extends Model
{
    use HasFactory;

    protected $table = 'transactions';
    public $timestamps = true;
    protected $primaryKey = 'id_transaction';

    protected $fillable = [
        'amount_diamond',
        'total_price',
        'status',
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id', 'user_id');
    }

    public function diamond()
    {
        return $this->belongsTo(Diamond::class, 'id_diamond', 'id_diamond');
    }
}
